From 030def406f8cac6a22837db2612e7b7c1049e6ab Mon Sep 17 00:00:00 2001
From: Martin Jansa <martin.jansa@lge.com>
Date: Tue, 4 Feb 2020 22:37:29 +0000
Subject: [PATCH] CMakeLists.txt: fix typo when searching for Python
 Interpreter

* causes whole CMake to fail with slightly misleading error:
ERROR: mavlink-2019.12.30-1-r0 do_configure: Function failed: do_configure (log file is located at /jenkins/mjansa/build-ros/webos-eloquent-warrior/tmp-glibc/work/raspberrypi4-webos-linux-gnueabi/mavlink/2019.12.30-1-r0/temp/log.do_configure.22693)
ERROR: Logfile of failure stored in: /jenkins/mjansa/build-ros/webos-eloquent-warrior/tmp-glibc/work/raspberrypi4-webos-linux-gnueabi/mavlink/2019.12.30-1-r0/temp/log.do_configure.22693
Log data follows:
| DEBUG: Executing shell function do_configure
| -- The C compiler identification is GNU 8.3.0
| -- The CXX compiler identification is GNU 8.3.0
| -- Check for working C compiler: /jenkins/mjansa/build-ros/webos-eloquent-warrior/tmp-glibc/work/raspberrypi4-webos-linux-gnueabi/mavlink/2019.12.30-1-r0/recipe-sysroot-native/usr/bin/arm-webos-linux-gnueabi/arm-webos-linux-gnueabi-gcc
| -- Check for working C compiler: /jenkins/mjansa/build-ros/webos-eloquent-warrior/tmp-glibc/work/raspberrypi4-webos-linux-gnueabi/mavlink/2019.12.30-1-r0/recipe-sysroot-native/usr/bin/arm-webos-linux-gnueabi/arm-webos-linux-gnueabi-gcc -- works
| -- Detecting C compiler ABI info
| -- Detecting C compiler ABI info - done
| -- Detecting C compile features
| -- Detecting C compile features - done
| -- Check for working CXX compiler: /jenkins/mjansa/build-ros/webos-eloquent-warrior/tmp-glibc/work/raspberrypi4-webos-linux-gnueabi/mavlink/2019.12.30-1-r0/recipe-sysroot-native/usr/bin/arm-webos-linux-gnueabi/arm-webos-linux-gnueabi-g++
| -- Check for working CXX compiler: /jenkins/mjansa/build-ros/webos-eloquent-warrior/tmp-glibc/work/raspberrypi4-webos-linux-gnueabi/mavlink/2019.12.30-1-r0/recipe-sysroot-native/usr/bin/arm-webos-linux-gnueabi/arm-webos-linux-gnueabi-g++ -- works
| -- Detecting CXX compiler ABI info
| -- Detecting CXX compiler ABI info - done
| -- Detecting CXX compile features
| -- Detecting CXX compile features - done
| CMake Error at /jenkins/mjansa/build-ros/webos-eloquent-warrior/tmp-glibc/work/raspberrypi4-webos-linux-gnueabi/mavlink/2019.12.30-1-r0/recipe-sysroot-native/usr/share/cmake-3.14/Modules/FindPackageHandleStandardArgs.cmake:211 (message):
|   No REQUIRED_VARS specified for FIND_PACKAGE_HANDLE_STANDARD_ARGS()
| Call Stack (most recent call first):
|   /jenkins/mjansa/build-ros/webos-eloquent-warrior/tmp-glibc/work/raspberrypi4-webos-linux-gnueabi/mavlink/2019.12.30-1-r0/recipe-sysroot-native/usr/share/cmake-3.14/Modules/FindPython/Support.cmake:1170 (find_package_handle_standard_args)
|   /jenkins/mjansa/build-ros/webos-eloquent-warrior/tmp-glibc/work/raspberrypi4-webos-linux-gnueabi/mavlink/2019.12.30-1-r0/recipe-sysroot-native/usr/share/cmake-3.14/Modules/FindPython3.cmake:181 (include)
|   CMakeLists.txt:27 (find_package)
|
|
| -- Configuring incomplete, errors occurred!
| See also "/jenkins/mjansa/build-ros/webos-eloquent-warrior/tmp-glibc/work/raspberrypi4-webos-linux-gnueabi/mavlink/2019.12.30-1-r0/build/CMakeFiles/CMakeOutput.log".
| WARNING: exit code 1 from a shell command.
| ERROR: Function failed: do_configure (log file is located at /jenkins/mjansa/build-ros/webos-eloquent-warrior/tmp-glibc/work/raspberrypi4-webos-linux-gnueabi/mavlink/2019.12.30-1-r0/temp/log.do_configure.22693)
NOTE: recipe mavlink-2019.12.30-1-r0: task do_configure: Failed
ERROR: Task (/jenkins/mjansa/build-ros/webos-eloquent-warrior/meta-ros/meta-ros2-eloquent/generated-recipes/mavlink/mavlink_2019.12.30-1.bb:do_configure) failed with exit code '1'

Signed-off-by: Martin Jansa <martin.jansa@lge.com>
---
 CMakeLists.txt | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 892d1f25..d8dcc0c4 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -24,14 +24,14 @@ else()
 endif()
 
 if (NOT CMAKE_VERSION VERSION_LESS "3.12")
-    find_package(Python${Py_VERSION} COMPONENTS Interpretator)
+    find_package(Python${Py_VERSION} COMPONENTS Interpreter)
 else()
     # find libraries with cmake modules
     find_package(PythonInterp ${Py_VERSION} REQUIRED)
     set(Python_EXECUTABLE "${PYTHON_EXECUTABLE}")
 endif()
 
-message(STATUS "Python ${Py_VERSION} iterpretator: ${Python_EXECUTABLE}")
+message(STATUS "Python ${Py_VERSION} iterpreter: ${Python_EXECUTABLE}")
 
 # Try to read package version from package.xml
 if(EXISTS ${CMAKE_SOURCE_DIR}/package.xml)
-- 
2.17.1

